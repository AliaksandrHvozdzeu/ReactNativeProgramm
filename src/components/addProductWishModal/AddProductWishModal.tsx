import React from 'react';
import {Image, Platform, Text, View} from 'react-native';
import {styles} from './styles';
import {Button} from 'react-native-elements';

type AddProductWishModalProps = {};

const SUCCESS_PNG_PATH: string = '../../assets/success.png';

const AddProductWishModal: React.FC<AddProductWishModalProps> = ({
  navigation,
}) => (
  <View style={styles.centeredView}>
    <View style={styles.modalView}>
      <View>
        <Image source={require(SUCCESS_PNG_PATH)} />
      </View>
      <View>
        <Text style={styles.modalText}>
          Product added to your wishlist successfully
        </Text>
      </View>
      <View>
        <Button
          buttonStyle={Platform.select({
            ios: styles.ios,
            android: styles.android,
          })}
          containerStyle={styles.containerStyle}
          onPress={() => navigation.goBack()}
          title="OK"
        />
      </View>
    </View>
  </View>
);

export default AddProductWishModal;
