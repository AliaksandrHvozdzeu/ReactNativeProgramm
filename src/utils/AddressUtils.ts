import {Component} from 'react';
import Geocode from 'react-geocode';
import {GOOGLE_MAP_API} from '@env';

class AddressUtils extends Component {
  static getBillingAddress(orderIncluded: []) {
    const zipcode = orderIncluded[4].attributes.zipcode;
    const countryIso3 = orderIncluded[4].attributes.country_iso3;
    const stateName = orderIncluded[4].attributes.state_name;
    const city = orderIncluded[4].attributes.city;
    const address1 = orderIncluded[4].attributes.address1;
    const address2 = orderIncluded[4].attributes.address2;
    const full: string = `${zipcode}, ${countryIso3}, ${stateName}, ${city}, ${address1}, ${address2}`;
    return full;
  }

  static getBillingCoordinatesByAddress(orderIncluded: []) {
    const zipcode = orderIncluded[4].attributes.zipcode;
    const city = orderIncluded[4].attributes.city;
    const address1 = orderIncluded[4].attributes.address1;
    const address2 = orderIncluded[4].attributes.address2;
    const full: string = `${zipcode}, ${city}, ${address1}, ${address2}`;

    Geocode.setApiKey(GOOGLE_MAP_API);
    Geocode.fromAddress(full).then(response => {
      const {lat, lng} = response.results[0].geometry.location;
      return {
        latitude: lat,
        longitude: lng,
        address: full,
      };
    });
  }
}

export default AddressUtils;
