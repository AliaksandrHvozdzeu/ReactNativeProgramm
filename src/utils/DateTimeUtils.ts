import {Component} from 'react';
import Moment from 'moment/moment';

const LOCALE: string = 'en';
const FORMAT: string = 'DD/MM/YY hh:mm';

class DateTimeUtils extends Component {
  static getFormatDate(date: string) {
    Moment.locale(LOCALE);
    return Moment(date).format(FORMAT);
  }
}

export default DateTimeUtils;
