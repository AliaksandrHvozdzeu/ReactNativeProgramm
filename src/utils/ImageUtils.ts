import {Component} from 'react';

class ImageUtils extends Component {
  static getImageById(imageId: string) {
    return `https://picsum.photos/id/${imageId}/3670/2462`;
  }
}

export default ImageUtils;
