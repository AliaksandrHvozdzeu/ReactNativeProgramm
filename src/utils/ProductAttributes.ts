import {Component} from 'react';

class ProductAttributes extends Component {
  static getSize(value: string) {
    if (!value) {
      return 'Size: N/A';
    }
    const myArray = value.split(',');
    return myArray[1].replace(' ', '');
  }

  static getExtraColor(value: string) {
    const myArray = value.split(',');
    if (myArray.length > 2) {
      return myArray[2].replace('and', '').replace('  ', '');
    }
  }

  static getColor(value: string) {
    if (!value) {
      return 'Color: N/A';
    }
    const myArray = value.split(',');
    return myArray[0];
  }
}

export default ProductAttributes;
