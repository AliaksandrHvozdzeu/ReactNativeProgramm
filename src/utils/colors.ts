export const COLORS = {
  blue_500: '#008ACE',
  blue_300: '#00A8F3',
  blue_100: '#8ED8F8',
  neutral_1000: '#1C1F27',
  neutral_700: '#4A4A4A',
  neutral_500: '#8F8F8F',
  neutral_300: '#C3C3C3',
  neutral_100: '#F7F7F7',
  neutral_0: '#FFFFFF',
  accent_orange_300: '#FEB96B',
  accent_orange_500: '#F39649',
  accent_red: '#DD6B55',
  accent_green: '#A5DC86',
  accent_green_dark: '#6AC238',
};
