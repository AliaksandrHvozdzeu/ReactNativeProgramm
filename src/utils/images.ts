export const ERROR_PNG_PATH = require('../assets/error.png');
export const PLUS_PNG_PATH = require('../assets/plus.png');
export const MINUS_PNG_PATH = require('../assets/minus.png');
export const SUCCESS_PNG_PATH = require('../assets/success.png');
export const WARNING_PNG_PATH = require('../assets/warning.png');
export const GOOD_PNG_PATH = require('../assets/good.png');
export const PERSON_PNG_PATH = require('../assets/person.png');
