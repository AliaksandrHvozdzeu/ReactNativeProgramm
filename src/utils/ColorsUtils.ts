import {Component} from 'react';
import {COLORS} from './colors';

class ColorsUtils extends Component {
  static getButtonColorTitle(title: string) {
    return title.replace('_', ' ');
  }

  static getButtonColor(color: string) {
    return {
      backgroundColor: color,
      borderRadius: 0,
      flex: 1,
      height: 30,
      width: 60,
    };
  }

  static getColorButtonTitleStyle(colorName: string) {
    return colorName === 'white'
      ? {
          color: COLORS.neutral_1000,
          fontSize: 10,
        }
      : {
          fontSize: 10,
        };
  }
}

export default ColorsUtils;
