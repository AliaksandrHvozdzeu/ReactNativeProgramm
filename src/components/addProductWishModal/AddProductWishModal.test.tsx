import 'react-native';
import React from 'react';
import {jest, test, describe, expect} from '@jest/globals';
import AddProductWishModal from './AddProductWishModal';
import {Image} from 'react-native';
import {SUCCESS_PNG_PATH} from '../../utils/images';
import {render} from '@testing-library/react-native';

const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

describe('AddProductWishModal tests', () => {
  test('Renders the AddProductWishModal correctly', () => {
    const tree = render(<AddProductWishModal />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Component AddProductWishModal contains image success image', () => {
    const wrapper = render(<AddProductWishModal />);
    const testInstance = wrapper.root;
    const textValue = testInstance.findByType(Image);
    expect(textValue.props.source).toBe(SUCCESS_PNG_PATH);
  });
});
