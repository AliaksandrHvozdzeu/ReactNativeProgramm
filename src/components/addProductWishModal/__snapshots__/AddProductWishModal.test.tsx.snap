// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProductWishModal tests Renders the AddProductWishModal correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 5,
        "elevation": 5,
        "margin": 20,
        "padding": 35,
        "shadowColor": "#1C1F27",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 4,
        "zIndex": 0,
      }
    }
  >
    <View>
      <Image
        alt="success_png_path"
        source="success.png"
      />
    </View>
    <View>
      <Text
        style={
          {
            "color": "#8F8F8F",
            "fontFamily": "Roboto",
            "fontSize": 20,
            "fontStyle": "normal",
            "fontWeight": "700",
            "lineHeight": 25,
            "marginTop": 10,
            "textAlign": "center",
          }
        }
      >
        Product added to your wishlist successfully
      </Text>
    </View>
    <View>
      <View
        style={
          [
            {
              "borderRadius": 3,
              "overflow": "hidden",
            },
            {
              "borderRadius": 3,
            },
            {
              "marginTop": 10,
              "width": 125,
            },
            false,
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": false,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#008ACE",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
                "shadowColor": "#4A4A4A",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 2,
                "shadowRadius": 4,
                "zIndex": 1,
              }
            }
          >
            <Text
              style={
                {
                  "color": "white",
                  "fontSize": 18,
                  "paddingVertical": 1,
                  "textAlign": "center",
                }
              }
            >
              OK
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
